qiskit_nature/second_q/circuit/library/ansatzes/puccsd.py

```diff
@@ -0,0 +1,172 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+The spin-adapted paired-UCC ansatz.
+"""
+
+from __future__ import annotations
+
+import logging
+from typing import Sequence, cast
+from collections import defaultdict
+
+from qiskit.circuit import QuantumCircuit
+from qiskit_nature.second_q.mappers import QubitConverter, QubitMapper
+from qiskit_nature.second_q.operators import FermionicOp
+
+from .ucc import UCC
+from .utils.fermionic_excitation_generator import (
+    generate_fermionic_excitations,
+    get_alpha_excitations,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PUCCSD(UCC):
+    """The spin-adapted paired-UCC Ansatz.
+
+    This ansatz (by default) contains paired single and double excitations. This ensures that not
+    only the number of particles but also the spin is preserved. [1]
+
+    Note, that this ansatz will produce a generalized operator pool (``generalized=True``).
+
+    This is a convenience subclass of the UCC ansatz. For more information refer to :class:`UCC`.
+
+    References:
+        [1] `arXiv:2207.00085 <https://arxiv.org/abs/2207.00085>`_
+
+    """
+
+    def __init__(
+        self,
+        num_spatial_orbitals: int | None = None,
+        num_particles: tuple[int, int] | None = None,
+        qubit_mapper: QubitConverter | QubitMapper | None = None,
+        *,
+        reps: int = 1,
+        initial_state: QuantumCircuit | None = None,
+    ) -> None:
+        # pylint: disable=unused-argument
+        """
+
+        Args:
+            num_spatial_orbitals: The number of spatial orbitals.
+            num_particles: The tuple of the number of alpha- and beta-spin particles.
+            qubit_mapper: The :class:`~qiskit_nature.second_q.mappers.QubitMapper` or
+                :class:`~qiskit_nature.second_q.mappers.QubitConverter` instance (use of the latter
+                is deprecated) which takes care of mapping to a qubit operator.
```

Comment:
With the typehint and the types listed in the opposite order when looking at the html, with the typehints standing out more I had to do a double take on this as at first it seemed the wrong way around with the latter reference.

![image](https://github.com/qiskit-community/qiskit-nature/assets/40241007/0bc50b12-03d1-40a7-a220-d9ca72882bf8)

I think I would almost argue with a new class just to document this for the QubitMapper - it may well work at present for QubitConvertor too while that code remains in the parent, but for "new code", to introduce with already deprecated  aspects...

Reply:
Ah this was a copy-paste mistake. I will remove `QubitConverter` :+1: